examples/routing.rs

```diff
@@ -180,43 +210,92 @@ fn main() {
                      .unwrap_or_else(|e| e.exit());
     if args.flag_help {
         println!("{:?}", args);
-        // return;
+        return;
     }
 
-    let test_node = RoutingNode::new(TestNode { stats: Arc::new(Mutex::new(Stats {stats: Vec::<(u32, TestData)>::new()})),
-                                                       ori_packets: Vec::<TestData>::new() });
-    let mutate_node = Arc::new(Mutex::new(test_node));
-    let copied_node = mutate_node.clone();
-    spawn(move || {
+    let mut command = String::new();
+    if args.flag_origin || args.flag_vault {
+        let test_node = RoutingNode::new(TestNode { stats: Arc::new(Mutex::new(Stats {stats: Vec::<(u32, TestData)>::new()})) });
+        let mutate_node = Arc::new(Mutex::new(test_node));
+        let copied_node = mutate_node.clone();
+        spawn(move || {
+            loop {
+                thread::sleep_ms(10);
+                copied_node.lock().unwrap().run();
+            }
+        });
+        if args.arg_endpoint.is_some() {
+            match SocketAddr::from_str(args.arg_endpoint.unwrap().trim()) {
+                Ok(addr) => {
+                    println!("initial bootstrapping to {} ", addr);
+                    let _ = mutate_node.lock().unwrap().bootstrap(Some(vec![Endpoint::Tcp(addr)]), None); 
+                }
+                Err(_) => {}
+            };
+        }
         loop {
-            thread::sleep_ms(10);
-            copied_node.lock().unwrap().run();
+            command.clear();
+            println!("Input command (stop, bootstrap <endpoint>)");
+            let _ = io::stdin().read_line(&mut command);
+            let v: Vec<&str> = command.split(' ').collect();
+            match v[0].trim() {
+                "stop" => break,
+                "bootstrap" => {
+                    let endpoint_address = match SocketAddr::from_str(v[1].trim()) {
+                        Ok(addr) => addr,
+                        Err(_) => continue
+                    };
+                    println!("bootstrapping to {} ", endpoint_address);
+                    let _ = mutate_node.lock().unwrap().bootstrap(Some(vec![Endpoint::Tcp(endpoint_address)]), None);
```

Comment:
@maqi  We always need to pass None for the bootstrap list option. Then only crust will try default methods like beacon.
If user provides an explicit bootstrap endpoint, then only we need to provide Some(Vec<Endpoint>) to the bootstrap method.

Refer crust documentation http://dirvine.github.io/crust/crust/connection_manager/struct.ConnectionManager.html#method.bootstrap
I will try to add more documentation to routing headers tomorrow to make this clear.